File: ALittle/Module/ALittleIDE/Script/UIEdit/IDEEditCore/IDESetting/ScrollScreenS.lua
-- ALittle Generate Lua
module("ALittleIDE", package.seeall)

local ___rawset = rawset
local ___pairs = pairs
local ___ipairs = ipairs


assert(ALittleIDE.DisplayLayoutS, " extends class:ALittleIDE.DisplayLayoutS is nil")
ScrollScreenS = Lua.Class(ALittleIDE.DisplayLayoutS, "ALittleIDE.ScrollScreenS")

function ScrollScreenS:Ctor(user_info, tab_child, tree_logic)
	___rawset(self, "_layer_name", "ide_setting_scrollscreen")
end

function ScrollScreenS:LoadNatureBase()
	DisplayLayoutS.LoadNatureBase(self)
	self:LoadShowTypeDataForExtends("container")
	self:LoadShowTypeDataForExtends("right_scrollbar")
	self:LoadShowTypeDataForExtends("bottom_scrollbar")
	self:LoadBoolData("open_extends_drag", true, g_IDEEnum.yn_type)
end

function ScrollScreenS:HandleContainerFOCUSOUT(event)
	if event.target._user_data ~= nil then
		if event.target._user_data == event.target.text then
			return
		end
		event.target._user_data = event.target.text
	end
	self:RemoverToNilShowSetForExtends("container", self._container.text, false)
end

function ScrollScreenS:HandleRightSFOCUSOUT(event)
	if event.target._user_data ~= nil then
		if event.target._user_data == event.target.text then
			return
		end
		event.target._user_data = event.target.text
	end
	self:RemoverToNilShowSetForExtends("right_scrollbar", self._right_scrollbar.text, false)
end

function ScrollScreenS:HandleBottomBFOCUSOUT(event)
	if event.target._user_data ~= nil then
		if event.target._user_data == event.target.text then
			return
		end
		event.target._user_data = event.target.text
	end
	self:RemoverToNilShowSetForExtends("bottom_scrollbar", self._bottom_scrollbar.text, false)
end

function ScrollScreenS:HandleHandOpenExtendsDragSELECT_CHANGE(event)
	self:BoolSelectChange("open_extends_drag", false, g_IDEEnum.yn_type)
end

File: ALittle/Module/ALittleIDE/JSScript/IDE/UIEdit/Setting/ScrollScreenS.js
{
if (typeof ALittleIDE === "undefined") window.ALittleIDE = {};


if (ALittleIDE.DisplayLayoutS === undefined) throw new Error(" extends class:ALittleIDE.DisplayLayoutS is undefined");
ALittleIDE.ScrollScreenS = JavaScript.Class(ALittleIDE.DisplayLayoutS, {
	Ctor : function(user_info, tab_child, tree_logic) {
		this._layer_name = "ide_setting_scrollscreen";
	},
	LoadNatureBase : function() {
		ALittleIDE.DisplayLayoutS.LoadNatureBase.call(this);
		this.LoadShowTypeDataForExtends("container");
		this.LoadShowTypeDataForExtends("right_scrollbar");
		this.LoadShowTypeDataForExtends("bottom_scrollbar");
		this.LoadBoolData("open_extends_drag", true, ALittleIDE.g_IDEEnum.yn_type);
	},
	HandleContainerFOCUSOUT : function(event) {
		if (event.target._user_data !== undefined) {
			if (event.target._user_data === event.target.text) {
				return;
			}
			event.target._user_data = event.target.text;
		}
		this.RemoverToNilShowSetForExtends("container", this._container.text, false);
	},
	HandleRightSFOCUSOUT : function(event) {
		if (event.target._user_data !== undefined) {
			if (event.target._user_data === event.target.text) {
				return;
			}
			event.target._user_data = event.target.text;
		}
		this.RemoverToNilShowSetForExtends("right_scrollbar", this._right_scrollbar.text, false);
	},
	HandleBottomBFOCUSOUT : function(event) {
		if (event.target._user_data !== undefined) {
			if (event.target._user_data === event.target.text) {
				return;
			}
			event.target._user_data = event.target.text;
		}
		this.RemoverToNilShowSetForExtends("bottom_scrollbar", this._bottom_scrollbar.text, false);
	},
	HandleHandOpenExtendsDragSELECT_CHANGE : function(event) {
		this.BoolSelectChange("open_extends_drag", false, ALittleIDE.g_IDEEnum.yn_type);
	},
}, "ALittleIDE.ScrollScreenS");

}